src/sentry/utils/eventuser.py

```diff
@@ -82,7 +86,11 @@ def get_display_name(self):
 
     @classmethod
     def for_projects(
-        self, projects: List[Project], keyword_filters: Mapping[str, Any]
+        self,
+        projects: List[Project],
+        keyword_filters: Mapping[str, List[Any]],
+        filter_boolean="AND",
+        return_all=False,
```

Comment:
The `return_all` parameter doesn't really work as we'd like it to at the moment. Previously, if we fetched all `EventUsers` for a project/set of keyword filters we'd receive one row per unique `EventUser`.

Since we're now fetching this from snuba, we're querying at the event level. So we'll actually return a row for every event that has an eventuser that matches the filters. This will end up returning duplicate rows, and if the `EventUser` data has changed slightly, say ip has changed, then we'll end up with slightly different duplicate rows as well.

We might need to use a window query here to work around this. https://clickhouse.com/docs/en/sql-reference/window-functions

Ideally we'd want to partition by the `sentry:user` tag and order by timestamp desc, then fetch the latest row for each unique tag. I'm not sure if the version of Clickhouse we're running supports window functions though, you'll need to check with SNS.

Another option is to just `GROUP BY` all the selected values. This should return something pretty close to what you want, and then you can just de-dupe the rest in code. Hard to say how well this works in practice in cases where ips are present, since that might cause a reasonable increase in rows.

Reply:
Ok, if you do this the it might be worth including `max(timestamp)` in the row as well. That way if you have two rows with the same `sentry:user` value you can choose the one wi the highest timestamp.

We should probably keep some stats on how many rows this returns.